### PR TITLE
Rearrange and reshuffle data documentation hub

### PR DESCRIPTION
**Pull request by jcjiang**

### CODE REVIEW

**Comment by jcjiang** [comment]
image
Rearranged content based on heatmap, removed repeats and replaced with scenario-based tutorials (for example, web API that accesses a database)

Changed titles to be "job-to-be-done" instead of the name of the tool - for example, "Accessing your data using C# (Entity Framework Core)" instead of "Entity Framework Core (Overview)"

File: entity-framework/dotnet-data/index.yml [hunk 1]
```diff
@@ -1,80 +1,81 @@
...
-    - title: "Azure Cosmos DB documentation"
-      itemType: reference
-      url: /azure/cosmos-db/
+    - title: "Tutorial: Access a SQLite database using C# (Entity Framework Core)"
+      itemType: quickstart
+      url: /ef/core/get-started/overview/first-app
 
     # Card
-    - title: "LINQ overview"
+    - title: "Using C# to work with a SQL Server database (Docs)"
```
Comment (ajcvickers) [inline entity-framework/dotnet-data/index.yml:43]: Mention Azure SQL as well

File: entity-framework/dotnet-data/index.yml [hunk 2]
```diff
@@ -1,80 +1,81 @@
...
-title: .NET data documentation
-summary: .NET is the open source and cross-platform solution for integrating data, data services, and data backends with your applications.
+title: .NET (C#) data documentation hub
+summary: This page contains resources about how to work with data using .NET (C#). Use our docs and tutorials to choose from our wide variety of available databases & data access approaches and get started quick!
 brand: dotnet
 
 metadata:
-  title: .NET data documentation
-  description: Welcome to the .NET data hub! What data-related task would you like to perform? This page is your gateway to documentation and tutorials for the various ways you interact with data from .NET. 
+  title:  .NET (C#) data documentation hub
```
Comment (ajcvickers) [inline entity-framework/dotnet-data/index.yml:8]: Does it add anything to specifically state "hub" here?

File: entity-framework/dotnet-data/index.yml [hunk 3]
```diff
@@ -1,80 +1,81 @@
...
       items:
         # Card
         - title: Relational databases
           links:
             - url: /dotnet/standard/data/sqlite/
-              text: "Microsoft.Data.Sqlite overview"
+              text: "Use .NET (C#) to connect and query data in SQLite"
             - url: /sql/
-              text: "Microsoft SQL documentation"
+              text: "Use .NET (C#) to connect and query data in Microsoft SQL Server"
```
Comment (ajcvickers) [inline entity-framework/dotnet-data/index.yml:74]: Azure SQL

File: entity-framework/dotnet-data/index.yml (hunk 2 above)
Comment (jcjiang) [reply entity-framework/dotnet-data/index.yml]: Not really, will take off!

**Pull request merged**